src/main/java/Ui.java

```diff
@@ -0,0 +1,46 @@
+public class Ui {
+    private TaskList taskList;
+
+    public Ui (TaskList taskList) {
+        this.taskList = taskList;
+    }
+
+    public void printList() {
+        System.out.println("Here are the tasks in your list!");
+        System.out.println("____________________________________________________________");
+        for (int i = 0; i < this.taskList.size(); i++) {
+            System.out.println("        " + (i + 1) + " " + this.taskList.get(i));
+        }
+        System.out.println("____________________________________________________________");
+    }
+
+    public void markAsDone(int index) {
+        this.taskList.get(index).markAsDone();
+        System.out.println("____________________________________________________________");
```

Comment:
maybe try repeating this line less by writing a separate function for it :) you seem to be repeating it in the majority of your functions in UI.java